diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b69782a23..80d17510b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,7 @@ include(CTest)
 
 # Disable Warnings as Errors for MSVC
 if (MSVC AND NOT CXX_CLANG)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3 /WX-")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4 /WX-")
 endif()
 
 if (PLATFORM_FREEBSD)
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 96ea429e5a..617d26d902 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -242,7 +242,7 @@ else()
   endif()
 endif()
 
-if(CXX_CLANG)
+if(CXX_CLANG AND NOT MSVC)
   target_compile_options(common PRIVATE -fsized-deallocation
                                         -Werror=unreachable-code-aggressive)
   target_compile_definitions(
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 62dab070e3..5d5eccebad 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1184,7 +1184,7 @@ else()
         -Wno-cast-function-type
 
         $<$<CXX_COMPILER_ID:Clang>:-fsized-deallocation>
-        $<$<CXX_COMPILER_ID:Clang>:-Wno-cast-function-type-mismatch>
+        $<$<AND:$<CXX_COMPILER_ID:Clang>,$<NOT:$<PLATFORM_ID:Android>>>:-Wno-cast-function-type-mismatch>
     )
 endif()
 
diff --git a/src/dynarmic/CMakeLists.txt b/src/dynarmic/CMakeLists.txt
index e5345ef458..c43afca4a9 100644
--- a/src/dynarmic/CMakeLists.txt
+++ b/src/dynarmic/CMakeLists.txt
@@ -66,7 +66,6 @@ if (MSVC)
         /experimental:external
         /external:W0
         /external:anglebrackets
-        /W4
         /w44263 # Non-virtual member function hides base class virtual function
         /w44265 # Class has virtual functions, but destructor is not virtual
         /w44456 # Declaration of 'var' hides previous local declaration
